compute_endpoint/globus_compute_endpoint/self_diagnostic.py

```diff
@@ -98,6 +98,19 @@ def kernel():
     return kernel
 
 
+def which(binary: str):
+    def kernel():
```

Comment:
I appreciate the "binary" no ... "string!" in the signature.  :tongue:

But less humorously, "binary" is, I think, a Windows-ism for "executable," but in Unix land they're a bit more precise with the term "executable."  For example, with a shebang, line, non-binary scripts (Bash, Dash, Perl, Python, Ruby, Tcl, ...) with the execute bit set (`chmod +x ...`) are considered "executable".

So, perhaps consider instead "exec_name", "exec_basename", "executable", "name", or something else slightly direct.